pennylane_lightning/src/gates/MyGateImplementation.hpp

```diff
@@ -0,0 +1,160 @@
+#include <vector>
+#include <thread>
+#include <algorithm>
+#include "BitUtil.hpp"
+#include "GateUtil.hpp"
+#include "Gates.hpp"
+#include "GateOperation.hpp"
+#include "KernelType.hpp"
+#include "PauliGenerator.hpp"
+
+using namespace Pennylane;
+using namespace Pennylane::Gates;
+
+class MyGateImplementation: public PauliGenerator<MyGateImplementation>{
+  public:
+    constexpr static KernelType kernel_id = KernelType::MyKernel; // Will be discussed below
+    constexpr static std::string_view name = "MyGateImpl"; // Name of your kernel
+
+    constexpr static std::array implemented_gates = {
+        GateOperation::PauliX,
+        GateOperation::PauliZ
+    }; // List of implemented gates
+
+
+    [[maybe_unused]] constexpr static std::array<GeneratorOperation,1> implemented_generators{};
+    
+    static inline 
+    std::vector<size_t> bounds(size_t parts, size_t mem){
+         std::vector<size_t> bnd;
+        size_t delta = mem/parts;
+        size_t reminder = mem%parts;
+        size_t N1=0, N2=0;
+        bnd.push_back(N1);
+        for(size_t i=0;i<parts;++i){
+            N2 = N1 + delta;
+            if(i == parts -1)
+                 N2+=reminder;
+            bnd.push_back(N2);
+            N1 = N2;
+        }
+        return bnd;
+    }
+
+
+    template <class PrecisionT>
+    static inline void PauliX_thread(std::complex<PrecisionT>* data, size_t L, size_t R, size_t wire_parity_inv, size_t wire_parity, size_t rev_wire_shift){
+        
```

Comment:
Given the codefactor complain here, we use `clang-tidy` and `clang-format` to ensure compliance of our code style. Running `make format` should automatically format your source to the repository style before committing.

Reply:
Thanks